l10n_nl_xaf_auditfile_export/models/xaf_auditfile_export.py

```diff
@@ -0,0 +1,253 @@
+        # namespaces, but we need the correct namespaces for validation
+        # we inject them at parse time in order not to traverse the document
+        # multiple times
+        default_namespace = 'http://www.auditfiles.nl/XAF/3.2'
+        iterparse = etree.iterparse(
+            StringIO(xml),
+            remove_blank_text=True, remove_comments=True)
+        for action, element in iterparse:
+            element.tag = '{%s}%s' % (default_namespace, element.tag)
+        del xml
+        xmldoc = etree.Element(
+            iterparse.root.tag,
+            nsmap={
+                None: default_namespace,
+                'xsi': 'http://www.w3.org/2001/XMLSchema-instance',
+            })
+        for element in iterparse.root:
+            xmldoc.append(element)
+        del iterparse
+
+        xsd = etree.XMLSchema(
+            etree.parse(
+                file(
+                    modules.get_module_resource(
+                        'l10n_nl_xaf_auditfile_export', 'data',
+                        'XmlAuditfileFinancieel3.2.xsd'))))
+        if not xsd.validate(xmldoc):
+            self.message_post('\n'.join(map(str, xsd.error_log)))
+            return
+
+        self.auditfile = base64.b64encode(etree.tostring(
+            xmldoc, xml_declaration=True, encoding='utf8'))
+
+    @api.multi
+    def get_odoo_version(self):
+        '''return odoo version'''
+        return release.version
+
+    @api.multi
+    def get_partners(self):
+        '''return a generator over partners and suppliers'''
+        offset = 0
+        while True:
+            results = self.env['res.partner'].search(
+                [
+                    '|',
+                    ('customer', '=', True),
+                    ('supplier', '=', True),
+                    '|',
+                    ('company_id', '=', False),
+                    ('company_id', '=', self.company_id.id),
+                ],
+                offset=offset,
+                limit=self.env['ir.config_parameter'].get_param(
+                    'l10n_nl_xaf_auditfile_export.max_records',
+                    default=MAX_RECORDS))
+            if not results:
+                break
+            offset += MAX_RECORDS
+            for result in results:
+                yield result
+            results.env.invalidate_all()
+            del results
+
+    @api.multi
+    def get_accounts(self):
+        '''return browse record list of accounts'''
+        return self.env['account.account'].search([
+            '|',
+            ('company_id', '=', False),
+            ('company_id', '=', self.company_id.id),
+        ])
+
+    @api.multi
+    def get_periods(self):
+        '''return periods in this export'''
+        return self.env['account.period'].search([
+            ('date_start', '<=', self.period_end.date_stop),
+            ('date_stop', '>=', self.period_start.date_start),
+            '|',
+            ('company_id', '=', False),
```

Comment:
I'm not so sure about that. We only export transactions, so this could mess up the balance I fear. The special periods are the ones where we make correction bookings, right?